kubeflow/katib/modeldb.libsonnet

```diff
@@ -0,0 +1,252 @@
+        },
+      },
+    },  // backendDeployment
+
+    dbService: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "db",
+        },
+        name: "modeldb-db",
+        namespace: namespace,
+      },
+      spec: {
+        ports: [
+          {
+            name: "dbapi",
+            port: 27017,
+            protocol: "TCP",
+          },
+        ],
+        selector: {
+          app: "modeldb",
+          component: "db",
+        },
+        type: "ClusterIP",
+      },
+    },  // dbService
+
+    dbDeployment: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "db",
+        },
+        name: "modeldb-db",
+        namespace: namespace,
+      },
+      spec: {
+        replicas: 1,
+        template: {
+          metadata: {
+            labels: {
+              app: "modeldb",
+              component: "db",
+            },
+            name: "modeldb-db",
+          },
+          spec: {
+            containers: [
+              {
+                image: "mongo:3.4",
+                name: "modeldb-db",
+                ports: [
+                  {
+                    containerPort: 27017,
+                    name: "dbapi",
+                  },
+                ],
+              },
+            ],
+          },
+        },
+      },
+    },  // dbDeployment
+
+    frontendService: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "frontend",
+        },
+        name: "modeldb-frontend",
+        namespace: namespace,
+      },
+      spec: {
+        ports: [
+          {
+            name: "api",
+            port: 3000,
+            protocol: "TCP",
+          },
+        ],
+        selector: {
+          app: "modeldb",
+          component: "frontend",
+        },
+        type: "ClusterIP",
+      },
+    },  // frontendService
+
+    frontendDeployment: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        labels: {
+          app: "modeldb",
+          component: "frontend",
+        },
+        name: "modeldb-frontend",
+        namespace: namespace,
+      },
+      spec: {
+        replicas: 1,
+        template: {
+          metadata: {
+            labels: {
+              app: "modeldb",
+              component: "frontend",
+            },
+            name: "modeldb-frontend",
+          },
+          spec: {
+            containers: [
+              {
+                args: [
+                  "modeldb-backend",
+                ],
+                env: [
+                  {
+                    name: "ROOT_PATH",
+                    value: "/katib",
+                  },
+                ],
+                image: "katib/katib-frontend",
+                imagePullPolicy: "Always",
```

Comment:
Default imagePullPolicy should be IfNotPresent. Always is not good practice; we want to use the local cache.